RangeControl: Convert component to TypeScript (#40535)
WordPress/gutenberg: packages/components/src/range-control/README.md

};
```

## Props

The set of props accepted by the component will be specified below.
Props not included in this set will be applied to the input elements.

### `afterIcon`: `string|Function|WPComponent|null`

> If this property is added, a DashIcon component will be rendered after the slider with the icon equal to afterIcon.

Reviewer: Nit: the `Icon` component can render more icon "types" than just `dashicon`s (also, I believe we're trying to move away from `dashicon`s in the repo)

```suggestion
If this property is added, an [Icon component](/packages/components/src/icon/README.md) will be rendered after the slider with the icon equal to afterIcon.
```

For more information on `IconType` see the [Icon component](/packages/components/src/icon/index.tsx#L23).

-   Required: No
-   Platform: Web

### `allowReset`: `boolean`

If this property is true, a button to reset the the slider is rendered.

-   Required: No
-   Default: `false`
-   Platform: Web | Mobile

### `beforeIcon`: `string|Function|WPComponent|null`

> If this property is added, a DashIcon component will be rendered before the slider with the icon equal to beforeIcon.

Reviewer: Same as previous comment:

```suggestion
If this property is added, an [Icon component](/packages/components/src/icon/README.md) will be rendered before the slider with the icon equal to beforeIcon.
```

For more information on `IconType` see the [Icon component](/packages/components/src/icon/index.tsx#L23).

-   Required: No
-   Platform: Web

### `color`: `CSSProperties['color']`

CSS color string for the `RangeControl` wrapper.

-   Required: No
-   Platform: Web

### `currentInput`: `number`

The current input to use as a fallback if `value` is currently `undefined`.

-   Required: No
-   Platform: Web

### `disabled`: `boolean`

Disables the `input`, preventing new values from being applied.

-   Required: No
-   Platform: Web

### `help`: `string|WPElement`

If this property is added, a help text will be generated using help property as the content.

-   Required: No
-   Platform: Web

### `hideLabelFromVision`: `boolean`

Provides control over whether the label will only be visible to screen readers.

- Required: No

### `icon`: `string`

> An icon to be shown above the slider next to it's container title.

Reviewer: ```suggestion
An icon to be shown above the slider next to its container title.
```
